scripts/releaseUtils.ts

```diff
@@ -0,0 +1,211 @@
+/**
+ * modified from https://github.com/vuejs/core/blob/master/scripts/release.js
+ */
+import colors from 'picocolors'
+import type { Options as ExecaOptions } from 'execa'
+import execa from 'execa'
+import { readFileSync, writeFileSync, existsSync, readdirSync } from 'fs'
+import path from 'path'
+import type { ReleaseType } from 'semver'
+import semver from 'semver'
+
+export const args = require('minimist')(process.argv.slice(2))
+
+export const isDryRun = !!args.dry
+
+if (isDryRun) {
+  console.log(colors.inverse(colors.yellow(' DRY RUN ')))
+  console.log()
+}
+
+export const packages = [
+  'vite',
+  'create-vite',
+  'plugin-legacy',
+  'plugin-react',
+  'plugin-vue',
+  'plugin-vue-jsx'
+]
+
+export const versionIncrements: ReleaseType[] = [
+  'patch',
+  'minor',
+  'major'
+  // 'prepatch',
+  // 'preminor',
+  // 'premajor',
+  // 'prerelease'
+]
+
+export function getPackageInfo(pkgName: string) {
+  const pkgDir = path.resolve(__dirname, '../packages/' + pkgName)
+
+  if (!existsSync(pkgDir)) {
+    throw new Error(`Package ${pkgName} not found`)
+  }
+
+  const pkgPath = path.resolve(pkgDir, 'package.json')
+  const pkg: {
+    name: string
+    version: string
+    private?: boolean
+  } = require(pkgPath)
+  const currentVersion = pkg.version
+
+  if (pkg.private) {
+    throw new Error(`Package ${pkgName} is private`)
+  }
+
+  return {
+    pkg,
+    pkgName,
+    pkgDir,
+    pkgPath,
+    currentVersion
+  }
+}
+
+export async function run(
+  bin: string,
+  args: string[],
+  opts: ExecaOptions<string> = {}
+) {
+  return execa(bin, args, { stdio: 'inherit', ...opts })
+}
+
+export async function dryRun(
+  bin: string,
+  args: string[],
+  opts?: ExecaOptions<string>
+) {
+  return console.log(
+    colors.blue(`[dryrun] ${bin} ${args.join(' ')}`),
+    opts || ''
+  )
+}
+
+export const runIfNotDry = isDryRun ? dryRun : run
+
+export function step(msg: string) {
+  return console.log(colors.cyan(msg))
+}
+
+export function getVersionChoices(currentVersion: string) {
+  const currentBeta = currentVersion.includes('beta')
+
+  const inc: (i: ReleaseType) => string = (i) =>
+    semver.inc(currentVersion, i, 'beta')!
+
+  const versionChoices = [
+    {
+      title: 'next',
+      value: inc(currentBeta ? 'prerelease' : 'patch')
+    },
+    ...(currentBeta
+      ? [
+          {
+            title: 'stable',
+            value: inc('patch')
+          }
+        ]
+      : [
+          {
+            title: 'beta-minor',
+            value: inc('preminor')
+          },
+          {
+            title: 'beta-major',
+            value: inc('premajor')
+          }
+        ]),
+    ...versionIncrements.map((i) => ({
+      value: inc(i),
+      title: i
+    })),
```

Comment:
Should we have the `minor`, `major` only when we aren't in beta? If the beta is on, we should only do a `next` or `release` it, no?
The `patch` isn't needed also, as it is the same as `next` if we aren't in beta. But we could leave it if you prefer that. 
```suggestion
    ...(currentBeta
      ? [
          {
            title: 'stable',
            value: inc('patch')
          }
        ]
      : [
          {
            title: 'beta-minor',
            value: inc('preminor')
          },
          {
            title: 'beta-major',
            value: inc('premajor')
          },
          {
            title: 'minor',
            value: inc('minor')
          },
          {
            title: 'major',
            value: inc('major')
          }
        ]),
```

Reply:
I meam that we dont need them always, we only should use them when we arent in the beta. And because they are two, we could remove the mapping